### PR TITLE
feat: auto set awq model_format from hf

### PR DESCRIPTION
**Pull request by zhyncs**
## Motivation

as titled

When the hf model is in awq format, there is no need to explicitly specify the model-format as awq.

```bash
# tested with https://huggingface.co/01-ai/Yi-6B-Chat-4bits
python3 -m lmdeploy serve api_server /workdir/Yi-6B-Chat-4bits
```

@lvhan028 @AllentDan May you help review this pr? Thanks.

## Modification

The modification is simple. I didn't write it as a function, but if needed, I am more than happy to make the changes.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit tests to ensure the correctness.
3. If the modification has a dependency on downstream projects of a newer version, this PR should be tested with all supported versions of downstream projects.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

**Review by AllentDan — APPROVED** [review]
LGTM

File: lmdeploy/turbomind/turbomind.py [hunk 1]
```diff
@@ -218,6 +219,15 @@ def _from_hf(self, model_source: ModelSource, model_path: str,
                 engine_config.model_format is None:
             engine_config.model_format = 'awq'
 
+        if engine_config.model_format is None:
+            cfg = get_hf_config_content(model_path)
+            quant_config = cfg.get('quantization_config')
+            if quant_config:
+                quant_method = quant_config.get('quant_method')
+                group_size = int(quant_config.get('group_size', 0))
+                if quant_method == 'awq' and group_size == 128:
```
Comment (lvhan028) [inline lmdeploy/turbomind/turbomind.py:228]: I think we should check the version attribute as well.
```
"version": "gemm",
```
There is another version `gemv` we haven't supported yet.

Comment (AllentDan) [reply lmdeploy/turbomind/turbomind.py:228]: Shall we give a warning or assertion error here if it does not meet lmdeploy's requirements?

Comment (zhyncs) [reply lmdeploy/turbomind/turbomind.py:228]: ok. I'll land the code.

Comment (zhyncs) [reply lmdeploy/turbomind/turbomind.py:228]: > it does not meet

Then we'll not set the model_format and at last it'll raise exception. So I think it's not a big deal if we don't address it here.

Comment (zhyncs) [reply lmdeploy/turbomind/turbomind.py]: > check the version attribute

@lvhan028 done

**Comment by zhyncs** [comment]
`pr_ete_test` is ImportError. Just ignore it.
> ImportError: This modeling file requires the following packages that were not found in your environment: flash_attn. Run `pip install flash_attn`

**Pull request merged**